### PR TITLE
Add programming tutorial

### PR DESCRIPTION
**Pull request by mildlyincompetent**

### CODE REVIEW

File: docs/programming.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+We can run it immediately. In order to have something we can run immediately in C++, we would have to write:
+
+```
+int main(){
+    doSomething();
+}
+```
+
+Note: just like with variables, you have to define the data type methods return, in the case of the `main` method this is an int.
+
```
Comment (antonnikitin97) [inline docs/programming.md]: they might write their functions, is it worth explaining the 'void' return type?

File: docs/programming.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+### Loops
+
+Both Python and C++ come with loops. The specific loops you're likely to find useful are **for loops** and **while loops**.
+
+#### For loops
```
Comment (antonnikitin97) [inline docs/programming.md]: #Pedantry - is for supposed to be capitalized?

File: docs/programming.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+Doing the same in C++ is a tad more complex:
+
+
+```
+for (int counter = 4; counter < 9; counter ++){
+    doSomething(counter);
+}
+```
+
+We see all the usual brackets adn semicolons, the bit that's particularly interesting is `int counter = 4; counter < 9; counter ++`. What's happening here is that we have 3 separate statements on a single line, the *initialisation* statement, a *condition* statement, and an *increment* statement.
```
Comment (antonnikitin97) [inline docs/programming.md]: > +We see all the usual brackets **adn** semicolons 

Spelling error.

File: docs/programming.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+The first statement of the loop (the initalisation statement) happens once, at the start of the loop. In our case, we declare the counter variable.
```
Comment (antonnikitin97) [inline docs/programming.md]: > op (the **initalisation** statement)

spelling error

File: docs/programming.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+```
+for (int counter = 4; counter < 9; counter ++){
+    doSomething(counter);
+}
+```
+
+We see all the usual brackets and semicolons, the bit that's particularly interesting is `int counter = 4; counter < 9; counter ++`. What's happening here is that we have 3 separate statements on a single line, the *initialisation* statement, a *condition* statement, and an *increment* statement.
+
+The first statement of the loop (the initalisation statement) happens once, at the start of the loop. In our case, we declare the counter variable.
```
Comment (antonnikitin97) [inline docs/programming.md]: statement of the loop (the 

> **initalisation** 

statement) ha

Spelling error

Comment (Adimote) [reply docs/programming.md]: fixed in 
88214e3

**Pull request merged**